src/adapter/src/rbac.rs

```diff
@@ -442,35 +459,34 @@ fn ownership_err(
     }
 }
 
+/// Generates the privileges required to execute a given plan.
+///
 /// The result of this function is a set of tuples of the form
 /// (What object the privilege is on, What privilege is required, Who must possess the privilege).
-///
-/// We use a [`BTreeSet`] because we may generate duplicate privileges while traversing through the
-/// plan and used objects.
 fn generate_required_privileges(
     catalog: &impl SessionCatalog,
     plan: &Plan,
     depends_on: &Vec<GlobalId>,
     role_id: RoleId,
-) -> BTreeSet<(ObjectId, AclMode, RoleId)> {
```

Comment:
We do still care about duplicates, we're just much more careful about not producing them (duplicates actually won't result in incorrectness just some wasted work). The main reason to switch to a `Vec` was because we want to maintain the insertion order of privileges so that the error messages come in a specific order. 

For example, when reading from a view, we want errors from the current user reading from the view to appear before errors from the view owner executing the view.

Let me know if that made any sense.